doc/user-guide.md

```diff
@@ -423,6 +423,88 @@ To implement complex deletion logic, you can add a finalizer to your Custom Reso
 deleted until you remove the finalizer (ie, after your cleanup logic has successfully run). For more information, see the
 [official Kubernetes documentation on finalizers](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#finalizers).
 
+**Example:**
+
+After the metadata name line your CR file add the reference to `finalizers`. If you are following the above example your CR file should look something like this:
+
+```sh
+$ cat deploy/crds/cache_v1alpha1_memcached_cr.yaml
+apiVersion: cache.example.com/v1alpha1
+kind: Memcached
+metadata:
+  name: example-memcached
+  finalizers:
+  - finalizer.cache.example.com
+spec:
+  # Add fields here
+  size: 3
+```
+
+Following is a snippet from the controller file under `pkg/controller/memcached/memcached_controller.go`
+
+```go
+func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+    reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+    reqLogger.Info("Reconciling Memcached")
+        
+    // Fetch the Memcached instance
+    memcached := &cachev1alpha1.Memcached{}
+    err := r.client.Get(context.TODO(), request.NamespacedName, memcached)
+
```

Comment:
Once you get the object, we need to add the finalizer if it is not there and no deletion timestamp.

Reply:
I will work on it.